### DISCUSSION TITLE
Extensions not loading

### DISCUSSION THREAD

**Discussion by naskio** [directus/directus#12160] [Archive]
Hello,
I am not able to load extensions to Directus v9.6.0.
I have an extension called 'example' in `extensions/endpoints/example/index.js`
```
"use strict";module.exports=e=>{e.get("/",((e,l)=>l.send("Hello, World!")))};
```
`GET /example` return this error
```
{"errors":[{"message":"Route /example doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}
```

I am currently using Docker (+ C ompose) with a mounted volume:

![Screen Shot 2022-03-15 at 08 56 15](https://user-images.githubusercontent.com/18899702/158331671-ca715daa-3a6a-4105-ab86-4ce520131d0d.png)

![Screen Shot 2022-03-15 at 08 53 51](https://user-images.githubusercontent.com/18899702/158331361-acb1d583-2a6b-4ed1-adce-086239c33227.png)

Am I missing something ?
Thanks for your help.

**Comment by rijkvanzanten** [comment]
Odd! I think it's how the extension is loaded into the docker container, rather than the extension itself. I just copy pasted your code into my extensions folder, and it works as expected:

![CleanShot 2022-03-15 at 10 57 45@2x](https://user-images.githubusercontent.com/9141017/158406363-27801d5b-9507-472b-9b12-68c5c7411408.png)

**Reply by naskio** [reply]
Hey
Thanks for your answer
Maybe there is a difference when in production mode (public url is set in my case).
I also checked if the folder exists inside the container and it is the case.
I will try to setup a local environment and test it.

**Reply by rijkvanzanten** [reply]
@naskio One quick way to make sure Directus is at least aware of your endpoint is by keeping an eye on stdout. It will log `✨ Loaded extensions: example` if Directus detected the extensions folder 👍🏻

**Comment by naskio** [comment]
Indeed, After trying to load them locally everything worked as expected:

![Screen Shot 2022-03-15 at 17 55 18](https://user-images.githubusercontent.com/18899702/158431998-ffd0c90d-3b9b-413b-a8e8-5cef525c1c91.png)

In prod, I kept getting this warning:
![Screen Shot 2022-03-15 at 18 02 20](https://user-images.githubusercontent.com/18899702/158432042-08903b2a-b9da-4f1c-9e69-4744b62ad061.png)

After investigating it, I found out that  the extension-folders (interfaces, layouts, modules, ...) need to exist to be able to load the extensions.

After creating the extension-folders, the extensions has been detected normally:

![Screen Shot 2022-03-15 at 18 10 46](https://user-images.githubusercontent.com/18899702/158433371-116bd76b-2dfc-44d4-aaed-f8f0919c03db.png)

![Screen Shot 2022-03-15 at 18 05 31](https://user-images.githubusercontent.com/18899702/158432555-22463c39-0824-4ebe-a187-550a17b5da7e.png)

**Reply by azrikahar** [reply]
Nice catch! Seems like it's the same "friction" faced in https://github.com/directus/directus/discussions/9955#discussioncomment-1677710. cc @nickrum

**Reply by staminna** [reply]
I have 4 different extensions loaded and I can see the lightning start emoji but none of them is actually changing anything in Directus.
Is there a menu I should click on a box inside the admin UI?

**Reply by br41nslug** [reply]
What kind of extensions @staminna ? Only the `module` type should directly change anything and might need to be enabled/shown under `Project Settings > Modules`.

**Reply by staminna** [reply]
@br41nslug I found out! It was on the display's link of the fields options! Thanks!

**Comment by dfanica** [comment]
Just to update this thread in case someone else stumbles across it... I was having a similar issue recently and it turns out the folder endpoints is no longer being used (or the syntax has changed and there's no documentation on this).

You now need to create a folder `directus-extension-{name}` and have the a `package.json` file with some required keys. I will link the documentation here: https://docs.directus.io/guides/extensions/endpoints-api-proxy-twilio.html

`.env` variables are will not work via `process.env` anymore, you now need to load it via the handler:

```
handler: (router, { services, env, logger }) => { ... }
```

I hope this helps someone. I was going mad after updating directus trying to figure out why my custom endpoints are not loading anymore.

P.S. If you're having issues with the `package.json` file you can always create an extension of type endpoint via `npm init directus-extension`; just make sure to remove the `"type" : "module"` from it afterwords as it doesn't like that apparently.

**Reply by br41nslug** [reply]
>  I was having a similar issue recently and it turns out the folder endpoints is no longer being used


https://docs.directus.io/releases/breaking-changes.html#deprecated-typed-extension-folders
